garak/generators/huggingface.py

```diff
@@ -53,10 +53,17 @@ def _set_hf_context_len(self, config):
                 self.context_len = config.n_ctx
 
     def _gather_hf_params(self, hf_constructor: Callable):
-        # this may be a bit too naive as it will pass any parameter valid for the pipeline signature
-        # this falls over when passed `from_pretrained` methods as the callable model params are not explicit
-        params = self.hf_args
-        if params["device"] is None:
+        """ "Identify arguments that impact huggingface transformers resources and behavior"""
+
+        # this may be a bit too naive as it will pass any parameter valid for the hf_constructor signature
+        # this falls over when passed some `from_pretrained` methods as the callable model params are not always explicit
+        params = (
+            self.hf_args
+            if hasattr(self, "hf_args") and isinstance(self.hf_args, dict)
+            else {}
+        )
```

Comment:
Do you have an example of where it falls over using `from_pretrained`? We could probably catch some of those.

Reply:
I will dig for an example, the comment above was due to some `from_pretrained()` methods being for the model and others for the tokenizer or other objects, and they take vastly different sets or config. Currently this method is only called to gather args for `transformers.AutoConfig.from_pretrained()`, `transformers.pipeline()`, and `transformers.LlavaNextForConditionalGeneration.from_pretrained()` initialization and we don't have a lot of examples of expanding beyond `device`, `device_map`, & `touch_dtype` as parameters to be passed on.

Also note the comment is not about this conditional, it is a general comment about the technique employed in this method possibly being brittle although I have not seen that much in practice yet, only during my iteration to get this technique working in the first place.